pkg/tuned/cmd/render/render.go

```diff
@@ -126,17 +129,40 @@ func render(inputDir []string, outputDir string) error {
 		}
 	}
 
-	perfProfilesTuneDProfiles := make(map[*performancev2.PerformanceProfile]*tunedv1.Tuned, len(perfProfiles))
-	for _, profile := range perfProfiles {
-		tunedFromPP, err := tuned.NewNodePerformance(profile)
-		if err != nil {
-			klog.Errorf("Unable to get tuned. error : %v", err)
-			return fmt.Errorf("unable to get tuned from profile: %w", err)
-		}
-		perfProfilesTuneDProfiles[profile] = tunedFromPP
-		// add tuned from PP to the list
-		tuneD = append(tuneD, tunedFromPP)
+	// Append any missing default manifests (i.e. `master`/`worker`)
+	mcPools = util.AppendMissingDefaultMCPManifests(mcPools)
+
+	mcp := findMachineConfigPoolByName(mcPools, mcpName)
+	if mcp == nil {
+		klog.Errorf("Unable to find MachineConfigPool:%q in input folders", mcpName)
+		return fmt.Errorf("Unable to find MachineConfigPool:%q in input folders", mcpName)
+	}
+
+	filteredPerformanceProfiles, err := filterPerformanceProfilesByMachineConfigPool(perfProfiles, mcp)
+	if err != nil {
+		klog.Errorf("Unable to find a PerformanceProfile that matches the MachineConfigPool %s. error : %v", mcpName, err)
+		return fmt.Errorf("Unable to get PerformanceProfile to apply using MachineConfigPool %s. error : %w", mcpName, err)
+	}
+
+	if len(filteredPerformanceProfiles) == 0 {
```

Comment:
We had just changed this behaviour in master with #924 so the absence of a performance profile is not considered an error.